src/Microsoft.AspNetCore.Server.Kestrel/Internal/Infrastructure/MemoryPool.cs

```diff
@@ -64,16 +65,33 @@ public class MemoryPool : IDisposable
         /// Called to take a block from the pool.
         /// </summary>
         /// <returns>The block that is reserved for the called. It must be passed to Return when it is no longer being used.</returns>
+#if DEBUG
+        public MemoryPoolBlock Lease([CallerMemberName] string memberName = "",
+                             [CallerFilePath] string sourceFilePath = "",
+                             [CallerLineNumber] int sourceLineNumber = 0)
```

Comment:
`Environment.StackTrace` caused the large response test to crash, which then caused a block to finalize. On the one hand it crashed; on the other it finalized a block - so may still be a circumstance...